### PR TITLE
EP/MIN-323 Reset liquidation attempts

### PR DESCRIPTION
**Pull request by Apollo50**
**Description:**

Add feature to reset liquidation attempts if fn deposit_underlying() called.
Fix unit test.
Fix benchmark.
Generate new weight file.

**Changes in storage or API:**

Describe storage structure changes or API changes. Make sure to update the [API docs](https://minterestfinance.atlassian.net/wiki/spaces/MINTEREST/pages/134807557/Minterest+API).

**Checklist**

- [x] Code is covered with tests
- [x] Code is covered with benchmarks
- [x] Fresh benchmark results included

### CODE REVIEW

File: pallets/minterest-protocol/src/lib.rs [hunk 1]
```diff
@@ -626,6 +626,12 @@ impl<T: Config> Pallet<T> {
 
 		T::MultiCurrency::deposit(wrapped_id, &who, wrapped_amount)?;
 
+		liquidity_pools::PoolUserParams::<T>::mutate(underlying_asset, &who, |p| {
+			if !p.liquidation_attempts.is_zero() {
+				p.liquidation_attempts = u8::zero();
+			}
```
Comment (dev-vadim) [inline pallets/minterest-protocol/src/lib.rs:632]: I don`t think we will benefit from this check.
```suggestion
			p.liquidation_attempts = u8::zero();
```

Comment (kurotych) [reply pallets/minterest-protocol/src/lib.rs:632]: agree

File: runtime/src/benchmarking/minterest_protocol.rs [hunk 2]
```diff
@@ -94,6 +94,9 @@ runtime_benchmarks! {
 		// set balance for lender
 		set_balance(DOT, &lender, 50_000 * DOLLARS)?;
 
+		// Set liquidation_attempts grater that zero to reset them.
```
Comment (lut1k) [inline runtime/src/benchmarking/minterest_protocol.rs:97]:
```suggestion
		// Set liquidation_attempts grater than zero to reset them.
```

**Review by lut1k — APPROVED** [review]

File: pallets/minterest-protocol/src/lib.rs (hunk 1 above)
Comment (Apollo50) [reply pallets/minterest-protocol/src/lib.rs:632]: Guys, One write = 1 000 000 000 weight, it's almost 1 millisecond :)
What is disadvantage of this check?

Comment (dev-vadim) [reply pallets/minterest-protocol/src/lib.rs:632]: I believe if we are already in mutate that is a write

Comment (dev-vadim) [reply pallets/minterest-protocol/src/lib.rs:632]: Maybe it`s worth checking by generating weights for both cases)

**Review by kurotych — APPROVED** [review]

File: pallets/minterest-protocol/src/lib.rs (hunk 1 above)
Comment (Apollo50) [reply pallets/minterest-protocol/src/lib.rs]: You were right, If we call mutate, then a write will be made.
What about this variant?

let mut pool_params = <LiquidityPools<T>>::pool_user_data(underlying_asset, &who);
if !pool_params.liquidation_attempts.is_zero() {
	pool_params.liquidation_attempts = u8::zero();
	liquidity_pools::PoolUserParams::<T>::insert(underlying_asset, &who, pool_params);
}

Comment (dev-vadim) [reply pallets/minterest-protocol/src/lib.rs]: Looks good for me

**Review by lut1k — DISMISSED** [review]

**Review by kurotych — APPROVED** [review]

**Review by lut1k — APPROVED** [review]

**Pull request merged**